packages/tests/components/SfRatingButton/SfRatingButton.PageObject.ts

```diff
@@ -0,0 +1,65 @@
+import { BasePage } from '../../utils/BasePage';
+
+export default class SfRatingButtonBaseObject extends BasePage {
+  hasValue(value: number) {
+    this.container.find('input:checked').should('have.attr', 'value', value);
+    return this;
+  }
+
+  hasAriaLabel(ariaLabel: string) {
+    this.container.should('have.attr', 'aria-label', ariaLabel);
+    return this;
+  }
+
+  allInputsHaveName(name: string, max: number) {
+    this.inputs.should(($inputs) => {
+      expect($inputs, `${max} inputs`).to.have.length(max);
+      Array.from({ length: max }, (_, index) => index).forEach((index) => {
+        expect($inputs.eq(index), `${index} input`).to.have.attr('name', name);
+      });
+    });
+    return this;
+  }
+
+  hasLabels(getLabelText: (value: number) => string, max: number) {
+    this.labels.should(($labels) => {
+      expect($labels, `${max} labels`).to.have.length(max);
+      Array.from({ length: max }, (_, index) => index + 1).forEach((value, index) => {
+        expect($labels.eq(index), `${index} input`).to.contain(getLabelText(value));
+      });
+    });
```

Comment:
That's an interesting take. I was thinking about whether we can create some nice util for this kind of cases, but I cannot think of anything nicer. So, as I don't have any strong opinion about it, let's leave it - good one!